### PR TITLE
BIP64 cleanup

### PR DESCRIPTION
**Pull request by dagurval**
This introduces the class `bip64::UTXORetriever` to encapsulate and remove duplicate logic between the network message `getutxos` and the rest call `get_utxos`.

In addition a qa test for BIP64 is added.

### CODE REVIEW

**Comment by dgenr8** [comment]
@xt-bot retest this

File: qa/rpc-tests/bip64.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        # Mine it, now it should exist in the chain.
+        xt_node.generate(1)
+        req_utxo(False)
+        assert_equal(test_node.utxos.bitmap[0], True)
+        assert_equal(test_node.utxos.result[0].height, xt_node.getblockcount())
+
+        # .. same result when including mempool
+        req_utxo(False)
+        assert_equal(test_node.utxos.bitmap[0], True)
+
```
Comment (dgenr8) [inline qa/rpc-tests/bip64.py]: Spent-in-mempool test needed.

**Comment by dgenr8** [comment]
@jli225 @shadders Have you got a Cash Lighthouse client you could point at this, to see how it works?

File: src/bip64_getutxo.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        Process(&viewChain, nullptr);
+    }
+}
+
+void UTXORetriever::Process(CCoinsView* view, CTxMemPool* mempool) {
+    for (const COutPoint& vOutPoint : outpoints) {
+        Coin coin;
+        bool hit = view->GetCoin(vOutPoint, coin);
+        if (mempool)
+            hit = !mempool->isSpent(vOutPoint);
```
[the remark this answers is not in the document]
Comment (dagurval) [reply src/bip64_getutxo.cpp]: Good catch

**Comment by dagurval** [comment]
Fixed issue found by @dgenr8 and added more tests

File: qa/rpc-tests/bip64.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        assert_equal(self.test_node.utxos.bitmap[0], int('0', 2))
+
+        # Mine tx creating new_outpoint, now it should exist in the chain.
+        xt_node.generate(1)
+        self.get_utxos([new_outpoint], checkmempool = False)
+        assert_equal(self.test_node.utxos.bitmap[0], int('1', 2))
+        assert_equal(self.test_node.utxos.result[0].height, xt_node.getblockcount())
+
+        # .. same result when including mempool
+        self.get_utxos([new_outpoint], checkmempool = False)
```
Comment (dgenr8) [inline qa/rpc-tests/bip64.py]: checkmempool = True here?
Testing that a bogus COutPoint appears in neither mode would also be nice.

**Comment by dgenr8** [comment]
@xt-bot retest this

**Pull request merged**